src/pages/Order/Order.js

```diff
@@ -1,16 +1,83 @@
 import React, { Component } from 'react';
 import CartList from './components/CartList';
-// import Favorit from './components/Favorit';
+import Favorit from './components/Favorit';
 import OrderHeader from './components/OrderHeader';
 import './Order.scss';
 
 export default class Order extends Component {
+  state = {
+    productInCart: [],
+  };
+
+  componentDidMount() {
+    const url = '/data/cart.json';
+    fetch(url)
+      .then(res => res.json())
+      .then(data =>
+        this.setState({
+          productInCart: data.cart_info,
+        })
+      );
+  }
+
+  handleCheckBox = e => {
+    const { productInCart } = this.state;
+    const arr = productInCart.map(item => {
+      if (item.name === e.target.value) {
+        return { ...item, is_checked: !item.is_checked };
+      } else {
+        return item;
+      }
+    });
+
+    this.setState({ productInCart: arr });
+  };
+
+  removeProduct = () => {
+    const { productInCart } = this.state;
+    this.setState({
+      productInCart: productInCart.filter(item => !item.is_checked),
+    });
+  };
+
+  clearCart = () => {
+    const { productInCart } = this.state;
+    // const option = {
+    //   method: 'POST',
+    //   headers: {
+    //     Authorization: localStorage.getItem('access_token'),
+    //   },
+    //   body: JSON.stringify({
+    //     order_id: order_id,
+    //   }),
+    // };
+
+    // fetch(CART_UPDATE_API, option)
+    //   .then(res => res.json())
+    //   .then(data => {
+    //     this.setState({
+    //       productInCart: [],
+    //     });
+    //   });
+
+    productInCart.length &&
+      this.setState({
+        productInCart: [],
+      });
```

Comment:
- 아직 주석 처리가 되어 있기는 하지만 장바구니를 clear 하는 함수 내에서 왜 조건에 따라 빈 배열을 setState 하는지 잘 이해가 되지 않습니다. 무조건 비워야 하지 않나요?
- 값의 선택이 아니라 로직상의 분기 처리를 할 때는 if 문을 사용하는 것이 의미상 좋아보입니다.
```suggestion
    if (productInCart.length) {
      this.setState({ productInCart: [] });
    }
```

Reply:
목데이터로 테스트 중이었어서 작성된 로직입니다.  통신후에는 필요없어 수정할 예정이에요.